research-app-messages/src/classic_pywwt.ts

```diff
@@ -187,15 +186,39 @@ export function isCreateAnnotationMessage(o: any): o is CreateAnnotationMessage
 }
 
 
-/** A command to create an image layer corresponding to a web-accessible FITS
- * file. */
-export interface CreateFitsLayerMessage {
+/** A command to create an image set layer. */
+export interface CreateImageSetLayerMessage {
   /** The tag identifying this message type. */
   event: "image_layer_create";
 
   /** An identifier for referring to this image layer later. */
   id: string;
 
+  /** The URL from which to obtain the FITS file or image set. */
+  url: string;
+
+  /** Tell WWT what type of layer you are Adding.
+   * OR let WWT try to autodetect the type of the data.
+   * Default, autodetect. */
+  mode: "autodetect" | "fits" | "preloaded";
+  }
+
+/** Type guard function for CreateImageSetLayerMessage. */
+export function isCreateImageSetLayerMessage(o: any): o is CreateImageSetLayerMessage {  // eslint-disable-line @typescript-eslint/no-explicit-any
+  return typeof o.event === "string" &&
+    o.event == "image_layer_create" &&
+    typeof o.id === "string" &&
+    typeof o.url === "string" &&
+    typeof o.mode  === "string";
```

Comment:
In order to avoid unsafety in the type system, here we need to check whether `mode` is explicitly one of the allowed string values (`"autodetect"` etc.), and not just any string.